src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -44,9 +50,26 @@ public class HttpFileSystemBasedFindPackageByIdResource : FindPackageByIdResourc
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="baseUris" /> is <c>null</c>.</exception>
         /// <exception cref="ArgumentException">Thrown if <paramref name="baseUris" /> is empty.</exception>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="httpSource" /> is <c>null</c>.</exception>
+        [Obsolete("Use constructor with source")]
         public HttpFileSystemBasedFindPackageByIdResource(
             IReadOnlyList<Uri> baseUris,
             HttpSource httpSource)
+            : this(source: null, baseUris, httpSource)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="HttpFileSystemBasedFindPackageByIdResource" /> class.
+        /// </summary>
+        /// <param name="baseUris">Base URI's.</param>
+        /// <param name="httpSource">An HTTP source.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="baseUris" /> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="baseUris" /> is empty.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="httpSource" /> is <c>null</c>.</exception>
+        public HttpFileSystemBasedFindPackageByIdResource(
+            string source,
```

Comment:
Can the `source` argument be obtained from `httpSource` argument ? (i.e. `httpSource.PackageSource.Source`)

Reply:
agree with @dominoFire  It's already been done that way anyways, and there's a null check on the package source param :)